treeherder/perf/alerts.py

```diff
@@ -120,3 +121,86 @@ def generate_new_alerts_in_series(signature):
                         'new_value': new_value,
                         't_value': t_value
                     })
+
+
+def os_relevance(alert):
+    platforms = [
+        {
+            'name': 'windows10',
+            'relevance': 5
+        },
+        {
+            'name': 'windows7',
+            'relevance': 4
+        },
+        {
+            'name': 'linux',
+            'relevance': 3
+        },
+        {
+            'name': 'osx',
+            'relevance': 2
+        },
+        {
+            'name': 'android',
+            'relevance': 1
+        }
+    ]
+
+    for platform in platforms:
+        if platform['name'] in alert.series_signature.platform.platform:
+            return platform['relevance']
+
+
+def select_from_each_platform(sorted_all_alerts, cherry_pick_config):
+    # Note: Ensure that the sorted_all_alerts container has only
+    # platforms of interest (example: 'windows10', 'windows7', 'linux', 'osx', 'android').
+    # Otherwise, the below 'while' will loop forever.
+    # In order to achieve this, please filter sorted_all_alerts list with filter_alerts(alerts) before calling this function.
+    circular_list = cycle(cherry_pick_config.platforms_of_interest)
+    platform_to_pick = next(circular_list)
+
+    picked_alerts = list()
+    current_alert = 0
+    while(current_alert < len(sorted_all_alerts)):
+        if platform_to_pick in sorted_all_alerts[current_alert].series_signature.platform.platform:
+            if len(picked_alerts) < cherry_pick_config.max_alerts_selected:
+                picked_alerts.append(sorted_all_alerts[current_alert])
+            current_alert += 1
+        platform_to_pick = next(circular_list)
+    return picked_alerts
+
+
+def filter_alerts(alerts, cherry_pick_config):
+    # Be aware only of mentioned platforms of interest
+    filtered_alerts = list()
+    for alert in alerts:
+        for platform in cherry_pick_config.platforms_of_interest:
+            if alert.series_signature.platform.platform.startswith(platform):
+                filtered_alerts.append(alert)
+    return filtered_alerts
+
+
+def ensure_alerts_variety(alerts, cherry_pick_config):
+    # Function intended to be used on already sorted alerts container.
+    for current_alert_index, alert in enumerate(alerts):
+        if not alert.is_regression and \
+                alerts[current_alert_index - 1].is_regression and \
+                current_alert_index > cherry_pick_config.max_alerts_selected:
+            alerts.insert(cherry_pick_config.max_alerts_selected - 1, alerts[current_alert_index])
+    return alerts
+
+
+class CherryPickConfig():
+    max_alerts_selected = 5
+    platforms_of_interest = ('windows10', 'windows7', 'linux', 'osx', 'android')
+
+
+def cherry_picked_alerts(alert_summary_id):
+    alert_summary = PerformanceAlertSummary.objects.get(id=alert_summary_id)
+    alerts = list(alert_summary.alerts.all())
```

Comment:
I think it would better suite us to extract lines 201-206 into a separate function, which we can then use inside this one.
The inner function would be more testable.

Reply:
A new function named extract_alert_list(alert_summary_id) will be responsible with DB query. Will do.